src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
I needed to set this for the test to work, because I'm querying random bogus fields. My guess is that the mappings are determined when first docs are indexed, because they have to take the values into account but we don't do this in the unit test setup. Will check this.